spec/controllers/books_controller_spec.rb

```diff
@@ -5,81 +5,159 @@
 RSpec.describe BooksController, type: :controller do
   describe 'GET #index' do
     subject { get :index }
-    it 'return all books' do
-      expect(response).to be_successful
+
+    let!(:user) { create(:user) }
+
+    context 'when user signin' do
+      before { sign_in user }
+
+      it 'return all books' do
+        expect(response).to be_successful
+      end
+    end
+
+    context 'when user not signin' do
+      it 'return status 302' do
+        expect(subject.status).to eq(302)
+        expect(subject).to redirect_to(new_user_session_path)
+      end
     end
   end
 
   describe 'GET #show' do
     subject { get :show, params: }
 
-    let(:book) { create(:book) }
+    let!(:user) { create(:user) }
+    let(:book) { create(:book, user_id: user.id) }
     let(:params) { { id: book.id } }
-    context 'when book existed' do
-      it 'return status 200' do
-        expect(subject.status).to eq(200)
-        expect(book).to eq(Book.find(book.id))
+
+    context 'when user signin' do
+      before { sign_in user }
+
+      context 'when Book existed' do
+        it 'return status 200' do
+          expect(subject.status).to eq(200)
+          expect(book).to eq(Book.find(book.id))
```

Comment:
We can use [this gem](https://github.com/rails/rails-controller-testing) to test if `@book` is being assigned to the correct record or not because `Book.find(book.id)` will be always equal to `book` anyway

```ruby
expect(assigns(:book)).to eq(book)
```